app/controllers/articles_controller.rb

```diff
@@ -1,4 +1,49 @@
 class ArticlesController < ApplicationController
   def index
+    @articles = Article.all
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+
+    if @article.save
+      redirect_to @article # use redirect_to when mutating app state, rather than render...
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def update
+    @article = Article.find(params[:id])
+    if @article.update(article_params)
+      redirect_to @article
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
+    @article.destroy
+
+    redirect_to root_path
+  end
+
+  private
+
+  def article_params
+    params.require(:article).permit(:title, :body)
   end
 end
```

Comment:
This is just a suggestion but to dry up your controller code you can create a private method 
`def set_article
      @article = Article.find(params[:id])
end`
then at the top add `before_action :set_article, only: [:show, :edit, :update, :destroy]`. And the result of doing all that is the you can remove `@article = Article.find(params[:id])` from show, edit, update, and destroy.

Reply:
I noticed that as well. It's a case of repetition of code which can be easily addressed.

Reply:
Ah, I didn't notice that, but it's so obvious when you point it out. I'll DRY it up. Thanks @J-pilon and @Noffica 😄